### PR TITLE
Avail server updates

### PR DESCRIPTION
**Pull request by richATavail**
Added Binary capabilities to Avail Server

### CODE REVIEW

File: avail-server/src/main/kotlin/com/avail/server/io/ServerInputChannel.kt [hunk 1]
```diff
@@ -150,7 +150,8 @@ class ServerInputChannel constructor(
 			markBuffer = null
 			if (waiters.isEmpty())
 			{
-				val message = Message(buffer.toString())
+				val message = Message(
+					buffer.toString().toByteArray(), channel.state)
```
Comment (richATavail) [inline avail-server/src/main/kotlin/com/avail/server/io/ServerInputChannel.kt]: @toddATavail I recall this feeling a bit weird. I'm going to look at it again...

Comment (richATavail) [reply avail-server/src/main/kotlin/com/avail/server/io/ServerInputChannel.kt]: Never mind, these are all CharBuffers that need to be made into Strings before they can be made into ByteArrays.

File: avail-server/src/main/kotlin/com/avail/server/io/WebSocketAdapter.kt [hunk 2]
```diff
@@ -2049,5 +2069,32 @@ class WebSocketAdapter @Throws(IOException::class) constructor(
...
+			reasonMessage: String,
+			reason: DisconnectReason)
+		{
+			val utf8 = StandardCharsets.UTF_8.encode(reasonMessage)
+			val buffer = ByteBuffer.allocateDirect(utf8.limit() + 2)
+			buffer.putShort(statusCode.statusCode.toShort())
+			buffer.put(utf8)
+			channel.channelCloseHandler.reason = reason
+			sendFrame(channel, Opcode.CLOSE, buffer)
+			{ channel.scheduleClose(CommunicationErrorDisconnect(it)) }
```
Comment (toddATavail) [inline avail-server/src/main/kotlin/com/avail/server/io/WebSocketAdapter.kt]: For safety, this probably needs to be reformatted:

```
sendFrame(channel, Opcode.CLOSE, buffer) {
	channel.scheduleClose(CommunicationErrorDisconnect(it))
}
```

Comment (richATavail) [reply avail-server/src/main/kotlin/com/avail/server/io/WebSocketAdapter.kt]: I made the requested change.

**Pull request merged**